edgepi-digital-out.py

```diff
@@ -6,27 +6,37 @@
 _logger=logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
-# arv[0] is this file's name, argv[1] should be numeric command
-if len(sys.argv) > 1:
+# create new DOUT instance
+try:
+    digital_output = EdgePiDigitalOutput()
+except ModuleNotFoundError as e:
+    _logger.error(f'Failed to load EdgePi modules: {e}')
+    sys.exit(0)
+
+# node channel property to SDK enum
+channelType_to_enum = {"dout1": GpioPins.DOUT1, "dout2": GpioPins.DOUT2, "dout3": GpioPins.DOUT3, "dout4": GpioPins.DOUT4,
+"dout5": GpioPins.DOUT5, "dout6":GpioPins.DOUT6, "dout7":GpioPins.DOUT7, "dout8": GpioPins.DOUT8}
+
+# keep receiving commands from parent process
+while True:
     try:
-        digital_output = EdgePiDigitalOutput()
-    except ModuleNotFoundError as e:
-        _logger.error(f'Failed to load EdgePi modules: {e}')
-        sys.exit(0)
-    
-    # keep receiving commands from parent process
-    while True:
-        try:
-            cmd = input()
-            # exit signal from parent process or user exit from child process
-            if 'exit' in cmd or cmd == '0':
-                sys.exit(0)
-            elif cmd == '5':
-                digital_output.digital_output_direction(GpioPins.DOUT4, False)
-                digital_output.digital_output_state(GpioPins.DOUT4, False)
-                print("Hello")
-        except(EOFError, SystemExit, KeyboardInterrupt):
-            sys.exit(0)
-else:
-    print("Number of commands must equal 2")
-    
\ No newline at end of file
+        # parse input from js file
+        inputs = input().split(",")
+        # get channel enum
+        channelType = inputs[0]
+        channel_enum = channelType_to_enum[channelType]
+        # get on/off
+        on_off = True if inputs[1].strip() == "true" else False
+
+        # action on dout based on input gathered
+        digital_output.digital_output_direction(channel_enum, False)
+        digital_output.digital_output_state(channel_enum, on_off)
+
+        # print message for payload
+        on_off_msg = "ON" if on_off else "OFF"
+        print(channelType.upper() + ": " + on_off_msg)
+
+        
+    # TO-DO : check exceptions
+    except(EOFError, SystemExit, KeyboardInterrupt, KeyError):
+        sys.exit(0)
```

Comment:
Not 100% sure what the TODO means but if it means to check the proper behavior of what should happen in each exception, then it's fine.

Reply:
Ran into some odd behavior and was coping with the KeyError exception and wanted to check it later. It's fixed now, so that comment is gone now. 88092acf17fe0a1aebb1a46fbb9f0d56583eb580